### PR TITLE
Upgrade + pin all dependencies

### PR DESCRIPTION
**Pull request by benoit74**
- Upgrade dependencies to latest version (especially python-scraper lib), including adapations for zimscraperlib 3
- Pin all dependencies as planned in our new rules

### CODE REVIEW

**Comment by benoit74** [comment]
I agree ! Could you explain what has to be done for "Advertise this new version" for scrappers ? A message on slack #scrappers ?

File: kolibri2zim/scraper.py
Comment (rgaudin) [inline kolibri2zim/scraper.py:781]: If you don't use those bytes…
```suggestion
            Illustration_48x48_at_1= self.favicon_fpath.read_bytes(),
```

**Review by rgaudin — CHANGES_REQUESTED** [review]
This is not a release PR ; this is a scraperlib upgrade one. Please remove the version from changelog and rename the PR accordingly.
Another change I'd like to see is using pinned versions for all dependencies. According to our [sacred book](https://github.com/openzim/_python-bootstrap/wiki/pyproject.toml#dependencies), we should stop using ranges for scrapers.

As @kelson42 mentioned, we usually create one such ticket per release, attached to the milestone it is supposed to end.
Release itself is usually composed of:
- A version change + changelog update commit on main:  release version and date at least but might include catch-up for changes that did not update the changelog
- A tag on that commit
- A Github release (manually created in UI). Some scrapers have been updated to trigger publication on Gihub `release` event, some are still using the previous behavior (on `tag` event)

Once this PR is merged, you can release yourself, directly on main, as you are the maintainer.

**Comment by rgaudin** [comment]
Oh and another step for scrapers is requesting me to update all ZF recipes to that new version.
If the release introduces CLI changes (of ZF interest, the `--nodes-id` one here we don't want to use on ZF), a PR on zimfarm must be created to update the definition obviously.

File: requirements.txt
Comment (benoit74) [inline requirements.txt:9]: @rgaudin is it correct, we have to pin all dependencies like this ?

Comment (rgaudin) [reply requirements.txt:9]: No;
Well that's debatable but the strategy behind [the doc](https://github.com/openzim/_python-bootstrap/wiki/pyproject.toml#dependencies) (I just added one line) is to only care about our own, direct dependencies.

We select them with care and when one of our dependency needs special attention (having a sub-dependency issue or using a range that we're afraid of), we pin the sub-dependency, with a comment stating why it's present.

`zimscraperlib` comes with many dependencies for instance. We only mention (and pin) it and it will pull `requests` (along with others). As soon as you'll be using `requests`, it will become a direct dependency and you'll pin it.

It doesn't require too much maintenance overhead and responsibilities are clear. PyPI prevents overwriting a version so there's no risk of change.
The risk lies in our direct dependencies definition. If they use ranges, there's a possibility for one of their dependency to change.

This is mitigated by:
- carefully choosing direct dependencies that care about their own.
- allowing ourselves to add and pin sub-dependencies
- tests whenever we can
- publishing docker images of all our releases.

I think it's a good compromise and should the need for more control arise (we've only add a couple issues with our wide ranges in years), we can still go beyond and freeze dependencies during the build process (thanks to hatch's flexibility, it should be easy) and/or use this lock file.

**Review by rgaudin — CHANGES_REQUESTED** [review]
LGTM ; except the dependencies ; see extensive inline comment.

File: requirements.txt
Comment (benoit74) [reply requirements.txt:9]: I had doubts. You convinced me ;)

**Pull request merged**
